src/MyFrame.cpp

```diff
@@ -648,11 +648,11 @@ MyFrame::HandlePuzException()
     }
     catch (puz::Exception & err)
     {
-        XWordMessage(this, MSG_PUZ_ERROR, (const wxChar *)puz2wx(puz::decode_utf8(err.what())).c_str());
+        XWordMessage(this, MSG_PUZ_ERROR, puz2wx(puz::decode_utf8(err.what())).c_str().AsChar());
     }
     catch (std::exception & err)
     {
-        XWordMessage(this, MSG_STD_EXCEPTION, (const wxChar *)puz2wx(puz::decode_utf8(err.what())).c_str());
+        XWordMessage(this, MSG_STD_EXCEPTION, puz2wx(puz::decode_utf8(err.what())).c_str().AsChar());

```

Comment:
The latter fails to compile in XCode: "Cannot pass object of non-POD type 'string_t' (aka 'basic_string<wchar_t>') through variadic function; call will abort at runtime".

(I'm also concerned about a similar issue to the one noted above - depending on whether #PUZ_UNICODE is set, the underlying representation is either char/string or wchar/wstring, which affects the format arg you need to use.)